Dockerfile

```diff
@@ -0,0 +1,213 @@
+# This dockerfile can be built in a few different ways.
+# Docker build commands must be run from within the base repository directory
+#
+# There are build args availalbe for specifying the:
+# - cq_version
+#   The version of CadQuery to use master or 2. 
+#   Default is 2.
+#   Options: [master, 2]
+#
+# - include_neutronics
+#   If software dependencies needed for neutronics simulations should be
+#   included true or false.
+#   Default is false.
+#   Options: [true, false]
+#
+# - compile_cores
+#   The number of CPU cores to compile the image with.
+#   Default is 1.
+#   Options: [1, 2, 3, 4, 5, 6...]
+#
+# Example builds:
+# Building using the defaults (cq_version 2, no neutronics and 1 core compile)
+# docker build -t ukaea/paramak .
+#
+# Building to include cadquery master, neutronics dependencies and use 8 cores.
+# Run command from within the base repository directory
+# docker build -t ukaea/paramak --build-arg include_neutronics=true --build-arg compile_cores=8 --build-arg cq_version=master .
+
+# Once build the dockerimage can be run in a few different ways.
+#
+# Run with the following command for a terminal notebook interface
+# docker run -it ukaea/paramak .
+#
+# Run with the following command for a jupyter notebook interface
+# docker run -p 8888:8888 ukaea/paramak /bin/bash -c "jupyter notebook --notebook-dir=/examples --ip='*' --port=8888 --no-browser --allow-root"
```

Comment:
**MAINTAINABILITY**

There's a failure launching Jupyter in Docker.

Some teams using this feature volume mount their own `/opt/notebooks` so there's not much of an issue. But maybe having some sample notebooks already in `/opt/notebooks` as a good stop-gap.

```
$ docker run -p 8888:8888 ukaea/paramak /bin/bash -c "jupyter notebook --notebook-dir=/opt/notebooks --ip='*' --port=8888 --no-browser --allow-root"
[C 21:15:45.561 NotebookApp] Bad config encountered during initialization: No such notebook dir: ''/opt/notebooks''
```
<p align="right">
    <a href="https://www.pullrequest.com/">
        <img width="15%" src="https://static.pullrequest.com/images/dashboard/pullrequest-logo.svg">
    </a>
</p>

Reply:
run command has been changed and simplified to no longer need the /bin/bash part